libraries/Microsoft.Bot.Builder/ActivityHandler.cs

```diff
@@ -434,13 +435,19 @@ protected virtual async Task<InvokeResponse> OnInvokeActivityAsync(ITurnContext<
         {
             try
             {
-                if (turnContext.Activity.Name == SignInConstants.VerifyStateOperationName || turnContext.Activity.Name == SignInConstants.TokenExchangeOperationName)
+                switch (turnContext.Activity.Name)
                 {
-                    await OnSignInInvokeAsync(turnContext, cancellationToken).ConfigureAwait(false);
-                    return CreateInvokeResponse();
-                }
+                    case SignInConstants.VerifyStateOperationName:
+                    case SignInConstants.TokenExchangeOperationName:
+                        await OnSignInInvokeAsync(turnContext, cancellationToken).ConfigureAwait(false);
+                        return CreateInvokeResponse();
+
+                    case "healthCheck":
+                        return CreateInvokeResponse(await OnHealthCheckAsync(turnContext, cancellationToken).ConfigureAwait(false));
```

Comment:
Consider breaking this into to lines, so folks stepping into the method have an easier time debugging.

Reply:
this is exactly the pattern we use all over the TeamsActivityHandler.
you can put a break point in the CreateInvokeResponse function if you like